joomla/joomla-cms
# [4.0] System - HTTP Headers - Write headers to the configuration file

### Summary of Changes

- Adds the new option `Write headers to the configuration file` (source https://github.com/zero-24/plg_system_httpheader/pull/6)
- restructure the plugin
- cleaup xml, doc blocks and php code

### Testing Instructions

#### Test  `Write headers to the configuration file`
- Install Joomla 4.0
- apply this patch
- go to the plugin config
- enable the new option `Write headers to the configuration file`
- save the plugin
- check the content of the .htaccess | web.config file
- change the static header configurations
- save the plugin
- check the .htaccess | web.config file

#### Test  the CSP Component Integration

- go to com_csp
- enable the detect option
- visit the backend and frontend
- notice the collected reports
- publish some of the reports
- enable the auto mode from the options (keep it report only for now :D)
- check the now generated csp header
- enable the custom mode
- configure a custom rule (adding script-src 'nonce-{nonce}' as minimum)
- check that `{nonce}` has been replaced by the actual nonce

### Expected result

The static headers are written to the respective server config file

### Actual result

This option does not exists

### Documentation Changes Required

- [ ] Update Screenshots (Plugin and Component options)
- [ ] Add new option description

https://docs.joomla.org/J4.x:Http_Header_Management
